source/base/quadrature_lib.cc

```diff
@@ -943,6 +943,8 @@ QSorted<dim>::QSorted(const Quadrature<dim> &quad) :
     {
       this->weights[i]           = quad.weight(permutation[i]);
       this->quadrature_points[i] = quad.point(permutation[i]);
+      if (permutation[i] != i)
+        this->is_tensor_product_flag = false;
```

Comment:
for clarity, it would probably be useful to set the flag to `true` just before the loop. Otherwise it looks like the variable may not actually be initialized (or at least, one has to look up what it is initialized to).